src/server/index.ts

```diff
@@ -208,6 +208,94 @@ export default (spreeClient: Instance, serverOptions: any) => {
       })
   })
 
+  app.post('/api/cart/apply-coupon', (request, response) => {
+    const { coupon } = request.query
+
+    spreeClient.cart.applyCouponCode(getTokenOptions(request), { coupon_code: coupon })
+      .then(() => {
+        logger.info(`Add coupon code = ${coupon}.`)
+
+        response.json({
+          code: 200,
+          result: true
+        })
+      })
+      .catch((error) => {
+        logger.error([`Could not add coupon code.`, error])
+        response.statusCode = 500
+        response.json({
+          code: 500,
+          result: null
+        })
+      })
+  })
+
+  app.post('/api/cart/shipping-information', (request, response) => {
+    logger.info('Fetching shipping information.')
+
+    const cartId = request.query.cartId
+    const extraParams = {
+      include: [
+        'line_items',
+        'line_items.variant',
+        'line_items.variant.product',
+        'line_items.variant.product.option_types'
+      ].join(',')
+    }
+
+    spreeClient.cart.show(getTokenOptions(request), extraParams)
+      .then((spreeResponse) => {
+        if (spreeResponse.isSuccess()) {
+          const successResponse: any = spreeResponse.success()
+          const resultAttr: any = successResponse.data.attributes
+          const lineItems = findIncludedOfType(successResponse, successResponse.data, 'line_items')
+          const items = lineItems.map((lineItem) => {
+            return getLineItem(successResponse, lineItem, cartId)
+          })
+
+          const result = {
+            totals: {
+              discount_amount: resultAttr.adjustment_total,
+              grand_total: resultAttr.total,
+              items_qty: resultAttr.items_qty,
+              quote_currency_code: resultAttr.currency,
+              shipping_amount: resultAttr.ship_total,
+              shipping_discount_amount: 0,
+              shipping_incl_tax: resultAttr.ship_total,
+              shipping_tax_amount: resultAttr.ship_total,
```

Comment:
`shipping_discount_amount `, `shipping_incl_tax ` and `shipping_tax_amount ` can all be removed.